src/StackAdmin/Azs.Compute.Admin/Module/Azs.Compute.Admin/Generated.PowerShell.Commands/SwaggerPathCommands/Add-AzsPlatformImage.ps1

```diff
@@ -121,95 +121,89 @@ function Add-AzsPlatformImage {
 
     Process {
 
-
         if ($PSCmdlet.ShouldProcess("$Publisher/$Offer/$Sku/$Version" , "Add new virtual machine image")) {
 
-            if ( [System.String]::IsNullOrEmpty($Location)) {
+            if ( -not $PSBoundParameters.ContainsKey('Location')) {
                 $Location = (Get-AzureRMLocation).Location
             }
-            $dontCheck = $true
-            try {
-                $result = Get-AzsPlatformImage -Publisher $Publisher -Offer $Offer -Sku $Sku -Version $Version -Location $Location
-                $dontCheck = $result -eq $null
-            } catch {
-                # No-op
-            }
 
-            if ($dontCheck -or ($Force.IsPresent -or $PSCmdlet.ShouldContinue("Virtual machine image exists with values $Publisher/$Offer/$Sku/$Version, overwrite?", "Performing operation add virtual machine image with publisher $Publisher, offer $Offer, SKU $Sku, and version $Version."))) {
```

Comment:
@deathly809 we should have the check for Force flag. I believe this will also prompt for overwrite if the image already exists. Any reason this is removed ?

Reply:
Fixed.

Reply:
@deathly809 force flag check needs to happen after the check for Get-AzsPlatformImage, if force flag is present, the SDK call should be made again to re-add the platform image

Reply:
We don't support force on Add/New.  That would be a Set-*.  Also, I think CARP throws an exception if you try to add an existing item such as PlatformImage/VMExtensions.

Reply:
Also, these are versioned, so you should never be able to overwrite an existing Image/Extension.